esp-hal/src/dma/mod.rs

```diff
@@ -1852,6 +1872,26 @@ pub trait DmaTxBuffer {
     fn length(&self) -> usize;
 }
 
+pub(crate) struct DmaTxBufferRef<'a, B: DmaTxBuffer> {
```

Comment:
You can use the same wrapper for both types if you do this.
```suggestion
pub(crate) struct DmaBufferRef<'a, B> {
```